qemu/tests/slof_device_tree.py

```diff
@@ -0,0 +1,80 @@
+import logging
+
+from autotest.client.shared import utils, error
+from virttest import utils_misc
+from virttest.qemu_devices import qcontainer
+from avocado.core import exceptions
+
+
+@error.context_aware
+def run(test, params, env):
+    """
+    slof_device_tree test:
+    steps:
+    1. Boot up a guest.
+    2. Check the guest device tree information
+    3. Compare the value with expected result.
+    :param test: Kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def get_info(vm_session, guest_info):
+        """
+        Check the corresponding information from vm.
+
+        :param vm_session: session to checked vm.
+        :return: corresponding prompt
+        """
+        (status, output) = vm_session.cmd_status_output(
+            "echo `cat /proc/device-tree/%s`" % guest_info)
+        if status != 0:
+            raise exceptions.TestFail("Failed to get %s" % guest_info)
+        return output.strip()
+
+    def check_nonexist_aliases(vm_session, devices):
+        """
+        Check a nonexist device aliases.
+
+        :param vm_session: session to checked vm.
+        :return: corresponding prompt
+        """
+
+        if devices.has_device("scsi-cd"):
+            logging.info("Skip nonexist_aliases check.")
+        else:
+            (status, output) = vm_session.cmd_status_output(
+                "echo `cat /proc/device-tree/aliases/cdrom`")
+            if status != 0:
+                logging.debug(output)
+            else:
+                raise exceptions.TestFail("Nonexist cdrom aliases check failed.")
+
```

Comment:
No step info in the case. It is very useful for tester to analyse the failed case. better to add as error_context.context("xxx", logging.info)